internal/clients/http/gcp/gcp_client.go

```diff
@@ -99,6 +109,31 @@ func (c *gcpClient) NewInstanceTemplatesClient(ctx context.Context) (*compute.In
 	return client, nil
 }
 
+func (c *gcpClient) GetServiceAccountEmail(ctx context.Context) (clients.GCPEmail, error) {
+	ctx, span := otel.Tracer(TraceName).Start(ctx, "GetServiceAccountEmail")
+	defer span.End()
+
+	client, err := c.newCloudManagerService(ctx)
+	if err != nil {
+		return "", fmt.Errorf("unable to create GCP cloud manager client: %w", err)
+	}
+
+	projectIamPolicy, err := client.Projects.GetIamPolicy(c.auth.Payload, &cm.GetIamPolicyRequest{}).Do()
+	if err != nil {
+		return "", fmt.Errorf("unable to get the project iam policy: %w", err)
+	}
+
+	for _, binding := range projectIamPolicy.Bindings {
+		if strings.Contains(binding.Role, "provisioning_role") {
+			if len(binding.Members) > 0 {
+				// The service account principal
+				return clients.GCPEmail(binding.Members[0]), nil
```

Comment:
Do I understand correctly that this will give us the first member of the group that:
1. needs to be called `provisioning_role`
2. we mandaded Customers to create and add our service account to it?

Would this actually result in "when following instructions correctly, returns our service account"?

My point:
1. if it's our service account we want, we probably don't need round trip to GCP
2. if it's our service account we return (and then compare with image.shared_with) how do we guide users that they need to share the image with this account (how would they know)?

Reply:
So first, it is just a proposal, like I explained on Slack it seems like we have a mismatch with image builder that we should fix. 
this proposal aims to return the service account principal which is defined in the customer account and has the "provisioning_role" in the role name (which is not great, we can change that)

> if it's our service account we want, we probably don't need round trip to GCP

I agree, another option of course is to avoid the round trip and just return our service account principal but I did not know if it is a good one because it is a hard-coded/config value 

> if it's our service account we return (and then compare with image.shared_with) how do we guide users that they need to share the image with this account (how would they know)?
This is the issue that we would need to resolve, one of the option is to ass our service account to be shared with the image. 
 I wanted you to see the UI works without problems when the image is shared with the service account for now.

Reply:
Let's just circut it and return our service account directly, without the GCP roundtrip and we can take it from there :)
I guess it should be easier solution, right?

Reply:
Looks like this was dropped from the patch. But leaving the merge on @ezr-ondrej thanks.